sycl/include/sycl/ext/oneapi/experimental/virtual_functions.hpp

```diff
@@ -60,16 +60,54 @@ struct PropertyMetaInfo<indirectly_callable_key::value_t<Set>> {
 #endif
 };
 
-template <typename First, typename... Rest>
-struct PropertyMetaInfo<calls_indirectly_key::value_t<First, Rest...>> {
-  static_assert(
-      sizeof...(Rest) == 0,
-      "assume_indirect_calls_to property only supports a single set for now");
+// Helper to concatenate several lists of characters into a single string.
```

Comment:
Considering that those helpers are only used during device compilation, I suggest that we put them all under `#ifndef __SYCL_DEVICE_ONLY__` to avoid going through unused code during device compilation pass

Reply:
Updated [7ac5a5e](https://github.com/intel/llvm/pull/15523/commits/7ac5a5e7956296ae33ec281170c598ceeae4def5)